src/matching/quoteMatcher.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ * QuoteMatcher matches quoted strings, respecting escaped quotes (\") and frieds
+ */
+export class QuoteMatcher {
+  static escapeChar = "\\";
+
+  static findOpening(char: string, corpus: string, start: number): number {
+    // first, search backwards to see if we could be inside a quote
+    for (let i = start - 1; i >= 0; i--) {
+      const currentChar = corpus[i];
+      if (currentChar === char && corpus[i - 1] !== this.escapeChar) {
+        return i;
+      }
+    }
+
+    // Didn't find one behind us, the string may start ahead of us
+    for (let i = start; i < corpus.length; i++) {
+      const currentChar = corpus[i];
+      if (currentChar === char && corpus[i - 1] !== this.escapeChar) {
```

Comment:
Correct, that is a limitation.

I looked at the algorithm used by other vim emulation stacks and this seemed to be the best one. I think we could handle that case by having another look-behind for another escape character? Or are there other pathological cases we'd still be missing?